account/account.go

```diff
@@ -316,103 +423,329 @@ func (account *Account) WaitForTransactionReceipt(ctx context.Context, transacti
 	return account.provider.BlockHashAndNumber(ctx)
 }
 
+// BlockNumber returns the block number of the account.
+//
+// Parameters:
+// - ctx: The context in which the function is called.
+// Returns:
+// - uint64: the block number as a uint64
+// - error: an error encountered
 func (account *Account) BlockNumber(ctx context.Context) (uint64, error) {
 	return account.provider.BlockNumber(ctx)
 }
 
+// BlockTransactionCount returns the number of transactions in a block.
+//
+// Parameters:
+// - ctx: The context.Context object for the function.
+// - blockID: The rpc.BlockID object representing the block.
+// Returns:
+// - uint64: the number of transactions in the block
+//  - error: an error, if any
 func (account *Account) BlockTransactionCount(ctx context.Context, blockID rpc.BlockID) (uint64, error) {
 	return account.provider.BlockTransactionCount(ctx, blockID)
 }
 
+// BlockWithTxHashes retrieves a block with transaction hashes.
+//
+// Parameters:
+// ctx - the context.Context object for the request.
+// blockID - the rpc.BlockID object specifying the block to retrieve.
+// Returns:
+// - interface{}: an interface{} representing the retrieved block
+// - error: an error if there was any issue retrieving the block
 func (account *Account) BlockWithTxHashes(ctx context.Context, blockID rpc.BlockID) (interface{}, error) {
 	return account.provider.BlockWithTxHashes(ctx, blockID)
 }
 
+// BlockWithTxs retrieves the specified block along with its transactions.
+//
+// Parameters:
+// - ctx: The context.Context object for the function.
+// - blockID: The rpc.BlockID parameter for the function.
+// Returns:
+// - interface{}: An interface{}
+// - error: An error
 func (account *Account) BlockWithTxs(ctx context.Context, blockID rpc.BlockID) (interface{}, error) {
 	return account.provider.BlockWithTxs(ctx, blockID)
 }
 
+// Call is a function that performs a function call on an Account.
+//
+// Parameters:
+// It takes in a context.Context object, a rpc.FunctionCall object, and a rpc.BlockID object as parameters.
+// Returns:
+// - []*felt.Felt: a slice of *felt.Felt
+// - error: an error object.
 func (account *Account) Call(ctx context.Context, call rpc.FunctionCall, blockId rpc.BlockID) ([]*felt.Felt, error) {
 	return account.provider.Call(ctx, call, blockId)
 }
 
+// ChainID returns the chain ID associated with the account.
+//
+// Parameters:
+// - ctx: the context.Context object for the function.
+// Returns:
+//   - string: the chain ID.
+//   - error: any error encountered while retrieving the chain ID.
 func (account *Account) ChainID(ctx context.Context) (string, error) {
 	return account.provider.ChainID(ctx)
 }
+
+// Class is a method that calls the `Class` method of the `provider` field of the `account` struct.
+//
+// Parameters:
+// - ctx: The context.Context
+// - blockID: The rpc.BlockID 
+// - classHash: The `*felt.Felt`
+// Returns:
+// - *rpc.ClassOutput: a `rpc.ClassOutput`
```

Comment:
The class output could be a DeprecatedContractClass or just a Contract class depending on the contract version